### PR TITLE
Add max block size xfield to rust-tapyrus

### PR DESCRIPTION
**Pull request by Naviabheeman**

### CODE REVIEW

File: src/blockdata/block.rs [hunk 1]
```diff
@@ -131,6 +141,7 @@ impl XField {
         match self {
             XField::None => 0u8,
             XField::AggregatePublicKey(_) => 1u8,
+            XField::MaxBlockSize(_) => 2u8,
```
Comment (azuchi) [inline src/blockdata/block.rs:144]: MaxBlockSize xfield value is 4 bytes, so why is it 9?

Comment (Naviabheeman) [reply src/blockdata/block.rs:144]: Sorry! I must have confused that its 8 byte integer. I'll check the tests and fix it

File: src/blockdata/block.rs [hunk 2]
```diff
@@ -455,6 +475,19 @@ mod tests {
...
+    fn xfield_max_block_size_test() {
+        let bytes = Vec::from_hex("0200010000").unwrap();
+        let decode: XField = deserialize(&bytes).unwrap();
+        assert_eq!(serialize(&decode), bytes);
+
+        let maxblocksize:u32 = 0x100;
+        assert_eq!(decode, XField::MaxBlockSize(maxblocksize));
+
+        let xfield = XField::from_str("0200010000");
+        assert_eq!(xfield.unwrap(), XField::MaxBlockSize(maxblocksize));
```
Comment (azuchi) [inline src/blockdata/block.rs:487]: Please add test cases for  boundary and abnormal values as well as normal values.

**Review by azuchi — CHANGES_REQUESTED** [review]

**Review by azuchi — APPROVED** [review]
